### PR TITLE
Add leftmost column with at least one solution

### PR DESCRIPTION
**Pull request by fbokovikov**

### CODE REVIEW

File: src/main/kotlin/leetcode/challenge/week3/LeftmostColumnWithAtLeastOne.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                end = middle
+                if (start == end) {
+                    return start
+                }
+            }
+        }
+        return -1
+    }
+
+    var min = n + 1
```
Comment (DmitryBokovikov) [inline src/main/kotlin/leetcode/challenge/week3/LeftmostColumnWithAtLeastOne.kt:42]: min = m;

File: src/main/kotlin/leetcode/challenge/week3/LeftmostColumnWithAtLeastOne.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    for (i in 0 until m) {
+        val curMin = binarySearch(i)
+        if (curMin >= 0) {
+            min = minOf(min, curMin)
+        }
+    }
+    return if (min == n + 1) -1 else min
```
Comment (DmitryBokovikov) [inline src/main/kotlin/leetcode/challenge/week3/LeftmostColumnWithAtLeastOne.kt:49]: min ==m ? -1 : min

File: src/main/kotlin/leetcode/challenge/week3/LeftmostColumnWithAtLeastOne.kt [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+fun leftMostColumnWithOneBS(binaryMatrix: BinaryMatrix): Int {
+    val dim = binaryMatrix.dimensions()
+    val m = dim[0]
+    val n = dim[1]
+
+    fun binarySearch(i : Int): Int {
+        var start = 0
+        var end = n - 1
+        while (start < end) {
```
Comment (DmitryBokovikov) [inline src/main/kotlin/leetcode/challenge/week3/LeftmostColumnWithAtLeastOne.kt:23]: please, get informed how to write binary search

File: src/main/kotlin/leetcode/challenge/week3/LeftmostColumnWithAtLeastOne.kt (hunk 2 above)
Comment (fbokovikov) [reply src/main/kotlin/leetcode/challenge/week3/LeftmostColumnWithAtLeastOne.kt:49]: There is no ternary operator in Kotlin. If you need one -> use if statement instead like in Python

File: src/main/kotlin/leetcode/challenge/week3/LeftmostColumnWithAtLeastOne.kt (hunk 1 above)
Comment (fbokovikov) [reply src/main/kotlin/leetcode/challenge/week3/LeftmostColumnWithAtLeastOne.kt:42]: It does not matter

File: src/main/kotlin/leetcode/challenge/week3/LeftmostColumnWithAtLeastOne.kt (hunk 3 above)
Comment (fbokovikov) [reply src/main/kotlin/leetcode/challenge/week3/LeftmostColumnWithAtLeastOne.kt:23]: That's fine way to use binary search.

What's wrong?

Comment (DmitryBokovikov) [reply src/main/kotlin/leetcode/challenge/week3/LeftmostColumnWithAtLeastOne.kt:23]: please pay attention how I write bs

File: src/main/kotlin/leetcode/challenge/week3/LeftmostColumnWithAtLeastOne.kt (hunk 2 above)
Comment (DmitryBokovikov) [reply src/main/kotlin/leetcode/challenge/week3/LeftmostColumnWithAtLeastOne.kt:49]: I meant another thing that's possible to find case when you sln doesn't work

**Pull request merged**